File: Algo_Ds_Notes/1-D_Array/Java/Mean_Variance_Standard_Deviation.java
import java.util.Scanner;
import java.util.*; 
import java.lang.Math;

public class Mean_Variance_Standard_Deviation
{
    static void meanVarianceStandardeviation(int arr[], int num) 
    {    
        float mean, v, sd;
        mean = 0;
        int i;

        for (i = 0; i < num; i++)
        {
            // Calculating sum of the array
            mean = mean + arr[i];
        }

        //  mean is the average of the numbers
        mean = mean / num;

        float sum = 0;

        for (i = 0; i < num; i++)
        {
            sum = sum + (float)(Math.pow(arr[i] - mean, 2));
        }

        // Variance is the average of the squared differences from the Mean
        v = sum / num;

        // Standard Deviation is a measure of how spread out numbers are, it is the square root of the Variance.
        sd = (float)Math.pow(v, 1/2);

        System.out.println("Mean = " + mean);
        System.out.println("Variance = " + v);
        System.out.println("Standard Deviation = " + sd);
    } 
    
    public static void main(String args[]) 
    {
        int n;
        Scanner s = new Scanner(System.in);
        n = s.nextInt();
        int arr[] = new int[n];
        for (int i = 0; i < n; i++)
        {
            arr[i] = s.nextInt();
        }

        meanVarianceStandardeviation(arr, n);
    }
}

/*
Input:
n = 5
arr = {1, 2, 3, 4, 5}
Output:
Mean = 3.0
Variance = 2.0
Standard Deviation = 1.0
*/

File: 1-D_Array/Go/Mean, Variance and Standard Deviation/Mean_Variance_Standard_Deviation.go
// Program in Go language to find mean , variance and standard deviation

package main

import (
	"fmt"
	"math"
)

func main() {

	// Declare the array size
	var arraySize int

	fmt.Print("Enter number of elements in your array: ")

	fmt.Scan(&arraySize)

	fmt.Println("Enter your array: ")

	// Create the array
	array := make([]int, arraySize)
	var mean, variance, sd, sum float64

	mean = 0
	for i := 0; i < arraySize; i++ {
		fmt.Scan(&array[i])

		// Calculating sum of the array
		mean = mean + float64(array[i])
	}

	//  mean is the average of the numbers
	mean = mean / float64(arraySize)

	// We are typecasting to float64 because, thats the type of argument given in math package
	sum = 0
	for i := 0; i < arraySize; i++ {
		sum = sum + math.Pow(float64(array[i])-mean, float64(2))
	}

	// Variance is the average of the squared differences from the Mean
	variance = float64(float64(sum) / float64(arraySize))

	// Standard Deviation is a measure of how spread out numbers are, it is the square root of the Variance.
	sd = math.Sqrt(variance)

	fmt.Println("\nMean = ", mean)
	fmt.Println("\nVariance = ", variance)
	fmt.Println("\nStandard Deviation = ", sd)

}

/*
Output after Execution:

Enter number of elements in your array: 6
Enter your array:
10
12
14
16
18
20

Mean =  15

Variance =  11.666666666666666

Standard Deviation =  3.415650255319866 */
